lib/kang.js

```diff
@@ -80,20 +85,42 @@ function knMakeSource(str)
 		str = str.substr(0, c);
 	}
 
-	c = str.indexOf(':');
-	if (c == -1) {
-		host = str;
+	if ((matched = IPV4_HOST.exec(str)) !== null) {
```

Comment:
Is there a standardized format for describing URLs that may contain either IPv4 or IPv6 addresses and ports?  I recall that at one time, Apache used square brackets around IPv6 addresses, but I don't know if that's a common convention, a standard, or what.  Either way, we should update the documentation here to reflect what's accepted.  This code has gotten a fair bit more complicated, and I wonder if we shouldn't be using a common library to parse this.

Within the implementation: was there a particular reason to replace the explicit parsing with regular expressions?